docs/OPENSHIFT_GUIDE.md

```diff
@@ -9,104 +10,248 @@
+```
 
-# Install KServe
-export KSERVE_VERSION=v0.10.1
-oc apply -f "https://github.com/kserve/kserve/releases/download/${KSERVE_VERSION}/kserve.yaml"
 
-# Patch the inferenceservice-config according to https://kserve.github.io/website/0.10/admin/serverless/kourier_networking/
-oc edit configmap/inferenceservice-config --namespace kserve
-# Add the flag `"disableIstioVirtualHost": true` under the ingress section
-ingress : |- {
-    "disableIstioVirtualHost": true
-}
-oc rollout restart deployment kserve-controller-manager -n kserve
-oc wait --for=condition=ready pod -l control-plane=kserve-controller-manager -n kserve --timeout=300s
+Next, follow the corresponding installation guide for the `Ingress` layer you want to use:
 
-# Install KServe built-in servingruntimes and storagecontainers
-oc apply -f "https://github.com/kserve/kserve/releases/download/${KSERVE_VERSION}/kserve-cluster-resources.yaml"
-```
+1. [Service Mesh](#installation-with-service-mesh)
+2. [Kourier](#installation-with-kourier)
 
 ## Installation with Service Mesh
 
+For more information about OpenShift ServiceMesh and its components like Jaeger and Kiali, please visit the [docs](https://docs.openshift.com/container-platform/4.17/service_mesh/v2x/installing-ossm.html).
+This is intended to be a very simple example and, we will install the Operator in the worker nodes, if you can to assign  
+the ServiceMesh to the `infrastructure` nodes, please see the documentation mentioned above. Also, only the ServiceMesh
+operator will be installed for simplicity and less resource consumption.
+
+Before continuing, make sure you have followed the steps in the [Installing OpenShift Serverless](#installing-openshift-serverless) section.
+
 ```bash
-# Install Service Mesh operators
+# Install Service Mesh operator
 oc apply -f openshift/service-mesh/operators.yaml
 oc wait --for=condition=ready pod --all -n openshift-operators --timeout=300s
 
+
 # Create an istio instance
-oc apply -f openshift/service-mesh/namespace.yaml
+oc create ns istio-system
 oc apply -f openshift/service-mesh/smcp.yaml
 oc wait --for=condition=ready pod --all -n openshift-operators --timeout=300s
 oc wait --for=condition=ready pod --all -n istio-system --timeout=300s
 
 # Make sure to add your namespaces to the ServiceMeshMemberRoll and create a
 # PeerAuthentication Policy for each of your namespaces
-oc create ns kserve 
-oc create ns kserve-demo
+oc create ns kserve; oc create ns kserve-demo
 oc apply -f openshift/service-mesh/smmr.yaml
 oc apply -f openshift/service-mesh/peer-authentication.yaml
 
-# Install OpenShift Serverless operator
-oc apply -f openshift/serverless/operator.yaml
-oc wait --for=condition=ready pod --all -n openshift-serverless --timeout=300s
-
 # Create an Knative instance
 oc apply -f openshift/serverless/knativeserving-istio.yaml
 
 # Wait for all pods in `knative-serving` to be ready
-oc get pod -n knative-serving
+watch -n2 oc get pods -n knative-serving
```

Comment:
`oc wait`?

Reply:
Yeah, I tried that first, but for a few seconds this command returns nothing until the pods are created, with the watch, at least you don't neet to submit the command again, but lets keep watch for consistency with the other ones.